server/boxd_api.py

```diff
@@ -121,7 +117,20 @@ def on_message(self, message_json):
                 # Todo:  come up with some error
                 return
 
-            if message['type'] == 'NICKNAME':
+            if message['type'] == 'JOIN_GAME':
+                name = message['data']['name']
+                assigned_game = GameRunner.assign_player(self.client_id)
+                GameRunner.update_player_name(self.client_id, name)
+
+                ConnectionManager.send_message(self.client_id, "You joined game {} as {}".format(assigned_game, name))
+                ConnectionManager.send_message(self.client_id, "The game has the following players:  {}"
+                                               .format(GameRunner().get_other_player_ids(self.client_id)))  # TODO:  Send player scores and board state
+                ConnectionManager.send_to_all(GameRunner.get_other_player_ids(self.client_id),
+                                              "{} (client {}) joined the game".format(name, self.client_id))
+
+            # TODO:  elif message['type'] == 'LEAVE_GAME':
```

Comment:
want to uncomment this and put a `print` placeholder so I can implement it anyways?

(as well as any other placeholders for events you're thinking of)